packages/block-editor/src/store/reducer.js

```diff
@@ -873,24 +865,22 @@ export const blocks = pipe(
 				const next = action.clientIds.reduce(
 					( accumulator, id ) => ( {
 						...accumulator,
-						[ id ]: reduce(
+						[ id ]: Object.entries(
 							action.uniqueByBlock
 								? action.attributes[ id ]
-								: action.attributes,
-							( result, value, key ) => {
-								// Consider as updates only changed values.
-								if ( value !== result[ key ] ) {
-									result = getMutateSafeObject(
-										state[ id ],
-										result
-									);
-									result[ key ] = value;
-								}
-
-								return result;
-							},
-							state[ id ]
-						),
+								: action.attributes
```

Comment:
This is probably a rare case, but the `attributes` can be `undefined`, and this will produce an error.

Test:

```
const blocks = wp.data.select('core/block-editor').getBlockOrder();
wp.data.dispatch('core/block-editor').updateBlockAttributes( blocks );
```

Reply:
Great find! I've added a test case for this scenario and added a fallback.

Let me know what you think!